File: TelerikUIExamples/TelerikUIExamples/PanZoom.m
//
//  PanZoom.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import <TelerikUI/TelerikUI.h>
#import "PanZoom.h"

@implementation PanZoom
{
    TKChart *_chart;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _chart = [[TKChart alloc] initWithFrame:[self exampleBounds]];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    _chart.allowPanDeceleration = YES;
    [self.view addSubview:_chart];
    
    NSMutableArray *items = [NSMutableArray new];
    for (int i = 0; i < 200; i++) {
        [items addObject:[[TKChartDataPoint alloc] initWithX:@(arc4random()%200) Y:@(arc4random()%1000)]];
    }
    TKChartScatterSeries *series = [[TKChartScatterSeries alloc] initWithItems:items];
    [_chart addSeries:series];
    
    items = [NSMutableArray new];
    for (int i = 0; i < 200; i++) {
        [items addObject:[[TKChartDataPoint alloc] initWithX:@(arc4random()%200) Y:@(arc4random()%1000)]];
    }
    series = [[TKChartScatterSeries alloc] initWithItems:items];
    [_chart addSeries:series];
    
    items = [NSMutableArray new];
    for (int i = 0; i < 200; i++) {
        [items addObject:[[TKChartDataPoint alloc] initWithX:@(arc4random()%200) Y:@(arc4random()%1000)]];
    }
    
    series = [[TKChartScatterSeries alloc] initWithItems:items];
    [_chart addSeries:series];
    
    _chart.xAxis.allowZoom = YES;
    _chart.xAxis.allowPan = YES;
    _chart.yAxis.allowZoom = YES;
    _chart.yAxis.allowPan = YES;
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/PanZoom.swift
//
//  PanZoom.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class PanZoom: TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        chart.allowPanDeceleration = true
        self.view.addSubview(chart)
        
        var items = [TKChartDataPoint]()
        for _ in 0..<200 {
            items.append(TKChartDataPoint(x: Int(arc4random() % 200), y: Int(arc4random() % 1000)))
        }
        var series = TKChartScatterSeries(items: items)
        chart.addSeries(series)
        
        items = [TKChartDataPoint]()
        for _ in 0..<200 {
            items.append(TKChartDataPoint(x: Int(arc4random() % 200), y: Int(arc4random() % 1000)))
        }
        series = TKChartScatterSeries(items: items)
        chart.addSeries(series)
        
        items = [TKChartDataPoint]()
        for _ in 0..<200 {
            items.append(TKChartDataPoint(x: Int(arc4random() % 200), y: Int(arc4random() % 1000)))
        }
        
        series = TKChartScatterSeries(items: items)
        chart.addSeries(series)
        
        chart.xAxis!.allowZoom = true
        chart.xAxis!.allowPan = true
        chart.yAxis!.allowZoom = true
        chart.yAxis!.allowPan = true
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

